integrations/integrations_test.go

```diff
@@ -0,0 +1,190 @@
+package integrations
+
+import (
+	"context"
+	"io"
+	"net"
+	"strings"
+	"testing"
+	"time"
+
+	"cosmossdk.io/log"
+	"cosmossdk.io/store/snapshots"
+	storetypes "cosmossdk.io/store/types"
+	abcitypes "github.com/cometbft/cometbft/abci/types"
+	bftclient "github.com/cometbft/cometbft/rpc/client/http"
+	bfttypes "github.com/cometbft/cometbft/types"
+	dbm "github.com/cosmos/cosmos-db"
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/server"
+	"github.com/cosmos/cosmos-sdk/server/api"
+	serverconfig "github.com/cosmos/cosmos-sdk/server/config"
+	servertypes "github.com/cosmos/cosmos-sdk/server/types"
+	"github.com/cosmos/gogoproto/grpc"
+	testapp "github.com/polymerdao/monomer/testapp"
+	"github.com/spf13/viper"
+	"github.com/stretchr/testify/require"
+)
+
+// Application Constructor `appCreator` for testing
+func mockAppCreator(
+	_ log.Logger,
+	db dbm.DB,
+	_ io.Writer,
+	_ servertypes.AppOptions,
+) servertypes.Application {
+	app, err := testapp.New(db, "1")
+	if err != nil {
+		panic(err)
+	}
+
+	return &WrappedTestApp{App: app}
+}
+
+// Unit test for Monomer's custom `StartCommandHandler` callback
+func TestStartCommandHandler(t *testing.T) {
+	svrCtx := server.NewDefaultContext()
+	svrCtx.Config.RootDir = t.TempDir()
+	svrCtx.Config.DBBackend = "memdb"
+
+	// This flag must be set, because by default it's set to ""
+	svrCtx.Viper.Set("minimum-gas-prices", "0.025stake")
+	// This flag must be set to load the Monomer genesis file
+	viper.Set(monomerGenesisPathFlag, "./testdata/genesis.json")
+	// This flag must be set to configure Monomer's Engine Websocket
+	viper.Set(monomerEngineWSFlag, "127.0.0.1:8089")
+
+	clientCtx := client.Context{}
+	inProcessConsensus := true
+	opts := server.StartCmdOptions{
+		DBOpener: func(rootDir string, backendType dbm.BackendType) (dbm.DB, error) {
+			return dbm.NewMemDB(), nil
+		},
+	}
+
+	cmtListenAddr := svrCtx.Config.RPC.ListenAddress
+	cmtListenAddr = strings.TrimPrefix(cmtListenAddr, "tcp://")
+
+	go func() {
+		err := StartCommandHandler(svrCtx, clientCtx, mockAppCreator, inProcessConsensus, opts)
+		require.NoError(t, err)
+	}()
+	time.Sleep(1 * time.Second)
+
+	// --- Ensure Monomer has an active CometBFT listener ---
+	if conn, err := net.Dial("tcp", cmtListenAddr); err != nil {
+		svrCtx.Logger.Error("failed to dial CometBFT", "error", err)
+		panic(err)
+	} else {
+		conn.Close()
+	}
```

Comment:
If we instead do something like `require.True(t, url.Parse(cmtListenAddr).IsReachable(ctx))`, we can avoid the sleep. Correct?

Reply:
Yes good point. This will allow us to avoid the sleep.